docs/src/main/sphinx/admin/properties-client-protocol.md

```diff
@@ -0,0 +1,183 @@
+# Client protocol properties
+
+The following sections provide a reference for all properties related to the
+[client protocol](/client/client-protocol). 
+
+(prop-protocol-spooling)=
+## Spooling protocol
+
+The following properties are related to the [](protocol-spooling).
+
+### `protocol.spooling.enabled`
+
+- **Type:** [](prop-type-boolean)
+- **Default value:** `true` (is that automatically configured if shares secret or so is set up)
+
+Enable spooled client protocol support. 
+
+### `protocol.spooling.shared-secret-key`
+
+- **Type:** [](prop-type-string)
+
+256 bit, base64-encoded secret key used to secure segment identifiers.
+
+### `protocol.spooling.retrieval-mode`
+    
+- **Type:** [](prop-type-string)
+- **Default value:** `STORAGE`
+
+Determines how the client retrieves the segment. Following are possible values:
+
+* `STORAGE` - client accesses the storage directly with the pre-signed URI, 1
+  round trip
+* `COORDINATOR_STORAGE_REDIRECT` - client accesses the coordinator and is
+  redirected to the storage with the pre-signed URI, 2 round trips
+* `COORDINATOR_PROXY` - client accesses the coordinator and gets segment data
+  through it, 1 round trip per segment 
+* `WORKER_PROXY` - tbd
+
+### `protocol.spooling.initial-segment-size`
+
+- **Type:** [](prop-type-data-size)
+- **Default value:** 8MB
+
+Initial size of the spooled segments.
+
+### `protocol.spooling.maximum-segment-size`
+
+- **Type:** [](prop-type-data-size)
+- **Default value:** 16MB
+
+Maximum size per segment.
+
+### `protocol.spooling.inlining.enabled`
+
+- **Type:** [](prop-type-boolean)
+- **Default value:** `true`
+
+Allow spooled protocol to inline data.
+
+
+### `protocol.spooling.inlining.max-rows`
+
+- **Type:** [](prop-type-integer)
+- **Default value:** 1000
+
+Maximum number of rows that are allowed to be inlined per worker.
+
+
+### `protocol.spooling.inlining.max-size`
+
+- **Type:** [](prop-type-data-size)
+- **Default value:** 128kB
+
+Maximum size of rows that are allowed to be inlined per worker.
```

Comment:
There are also
```
protocol.spooling.encoding.json.enabled
protocol.spooling.encoding.json+zstd.enabled
protocol.spooling.encoding.json+lz4.enabled
```

all true by default, that allows to disable specific encodings server side.